lib/libc/stdlib/getenv.c

```diff
@@ -448,6 +448,17 @@ getenv(const char *name)
 }
 
 
+/*
+ * Runs getenv() unless the current process is tainted by uid or gid changes, in which case it will return NULL.
+ */
+char *
+secure_getenv(const char *name)
+{
+    if (issetugid())
```

Comment:
Any other conditions that should return null? The linux man page returns null for other conditions...

Reply:
If you adopt @mjguzik 's man page suggestions, this becomes a non-issue.